File: CPPVersion/AoC2020CPP/AoC2020CPP/Day3.cpp
#include "Day3.h"
#include <iostream>

Day3::Day3() :DefaultClass(3, true)
{
	inp = FR.ReadLine();
	ResetClock();
	std::cout << "Part1: " << SlopValue(3, 1) << "\n";
	TimeClock();
	ResetClock();
	long long sum = SlopValue(1, 1);
	sum *= SlopValue(3, 1);
	sum *= SlopValue(5, 1);
	sum *= SlopValue(7, 1);
	sum *= SlopValue(1, 2);
	std::cout << "Part2: " << sum << "\n";
	TimeClock();
}

long Day3::SlopValue(int right, int down)
{
	int sum = 0;
	int x=0;
	for (int i = 0; i < inp.size(); i+=down)
	{
		if (inp[i][x] == '#')
			sum++;
		x = (x+right) % inp[0].length();
	}
	return (long)sum;
}

File: JavaVersion/AoC2020/src/com/company/Day3.java
package com.company;

import java.util.ArrayList;

public class Day3 extends DefaultClass {

    public Day3()
    {
        super(3,true);
        ArrayList<String> Input = fr.ReadLine();
        ResetClock();
        System.out.println("Part1: " + TreeEncounter(3,1,Input));
        TimeClock();
        ResetClock();
        long sum = TreeEncounter(1,1,Input);
        sum*=TreeEncounter(3,1,Input);
        sum*=TreeEncounter(5,1,Input);
        sum*=TreeEncounter(7,1,Input);
        sum*=TreeEncounter(1,2,Input);
        System.out.println("Part2: "+sum);
        TimeClock();
    }

    private int TreeEncounter(int right, int down,ArrayList<String> Input)
    {
        int x=0;
        int sum=0;
        for (int i = 0; i < Input.size(); i+=down) {
            if(Input.get(i).charAt(x)=='#')
                sum++;
            x=(x+right)%Input.get(0).length();
        }
        return sum;
    }
}
